exercises/nodejs/src/game.js

```diff
@@ -1,172 +1,206 @@
-    if (places[currentPlayer] == 8) { return 'Pop' }
-    if (places[currentPlayer] == 1) { return 'Science' }
-    if (places[currentPlayer] == 5) { return 'Science' }
-    if (places[currentPlayer] == 9) { return 'Science' }
-    if (places[currentPlayer] == 2) { return 'Sports' }
-    if (places[currentPlayer] == 6) { return 'Sports' }
-    if (places[currentPlayer] == 10) { return 'Sports' }
+    if (playerPositions[currentPlayer] % 4 == 0) { return 'Pop' }
+    if (playerPositions[currentPlayer] % 4 == 1) { return 'Science' }
+    if (playerPositions[currentPlayer] % 4 == 2) { return 'Sports' }
     return 'Rock'
   }
 
-  this.createRockQuestion = function (index) {
-    return 'Rock Question ' + index
+  this.createQuestion = function (category, index) {
+    return category + ' Question ' + index
   }
 
-  for (var i = 0; i < 50; i++) {
-    popQuestions.push('Pop Question ' + i)
-    scienceQuestions.push('Science Question ' + i)
-    sportsQuestions.push('Sports Question ' + i)
-    rockQuestions.push(this.createRockQuestion(i))
+  this.createQuestionSet = function () {
+    for (var i = 0; i < NumberOfQuestions; i++) {
+      for(let category of Object.keys(questionCategories)) {
+        questionCategories[category].push(this.createQuestion(category, i))
+      }
+    }
   }
 
-  this.isPlayable = function (howManyPlayers) {
+  this.canPlayGame = function (howManyPlayers) {
     return howManyPlayers >= 2
   }
 
-  this.add = function (playerName) {
-    players.push(playerName)
-    places[this.howManyPlayers() - 1] = 0
-    purses[this.howManyPlayers() - 1] = 0
-    inPenaltyBox[this.howManyPlayers() - 1] = false
+  this.addPlayer = function (playerName) {
+    playerNames.push(playerName)
+    playerPositions[this.numberOfPlayers() - 1] = 0
+    playerPurses[this.numberOfPlayers() - 1] = 0
+    inPenaltyBox[this.numberOfPlayers() - 1] = false
 
     console.log(playerName + ' was added')
-    console.log('They are player number ' + players.length)
-
+    console.log('They are player number ' + playerNames.length)
     return true
   }
 
-  this.howManyPlayers = function () {
-    return players.length
+  this.numberOfPlayers = function () {
+    return playerNames.length
   }
 
-  var askQuestion = function () {
-    if (currentCategory() == 'Pop') { console.log(popQuestions.shift()) }
-    if (currentCategory() == 'Science') { console.log(scienceQuestions.shift()) }
-    if (currentCategory() == 'Sports') { console.log(sportsQuestions.shift()) }
-    if (currentCategory() == 'Rock') { console.log(rockQuestions.shift()) }
+  var askQuestion = function (category) {
+    console.log(questionCategories[category].shift())
   }
 
   this.roll = function (roll) {
-    console.log(players[currentPlayer] + ' is the current player')
+    console.log(playerNames[currentPlayer] + ' is the current player')
     console.log('They have rolled a ' + roll)
 
     if (inPenaltyBox[currentPlayer]) {
       if (roll % 2 != 0) {
```

Comment:
You could potentially create a method `isOddRoll(roll)` to remove the 'primitive conditional from this statement, and make it read more like a sentence:

```suggestion
      if (isOddRoll(roll)) { ... }
      
      function isOddRoll(roll) {
          return roll % 2 != 0
      }
```

Reply:
Thank you, this is a good idea, I will implement locally